docs/src/main/mdoc/consumers.md

```diff
@@ -0,0 +1,314 @@
+    }
+
+    def configure(configs: java.util.Map[String, _], isKey: Boolean): Unit = ()
+    def close(): Unit = ()
+  }
+}.suspend
+```
+
+Note that `close` and `configure` won't be called for the delegates.
+
+## Settings
+
+In order to create a [`KafkaConsumer`][kafkaconsumer], we first need to create [`ConsumerSettings`][consumersettings]. At the very minimum, settings include the effect type to use, and the key and value deserializers. More generally, [`ConsumerSettings`][consumersettings] contain everything necessary to create a [`KafkaConsumer`][kafkaconsumer]. If deserializers are available implicitly for the key and value type, we can use the syntax in the following example.
+
+```scala mdoc:silent
+val consumerSettings =
+  ConsumerSettings[IO, String, String]
+    .withAutoOffsetReset(AutoOffsetReset.Earliest)
+    .withBootstrapServers("localhost")
+    .withGroupId("group")
+```
+
+We can also specify the deserializers explicitly when necessary.
+
+```scala mdoc:silent
+ConsumerSettings(
+  keyDeserializer = Deserializer[IO, String],
+  valueDeserializer = Deserializer[IO, String]
+).withAutoOffsetReset(AutoOffsetReset.Earliest)
+ .withBootstrapServers("localhost")
+ .withGroupId("group")
+```
+
+[`ConsumerSettings`][consumersettings] provides functions for configuring both the Java Kafka consumer and options specific to the library. If functions for configuring certain properties of the Java Kafka consumer is missing, we can instead use `withProperty` or `withProperties` together with constants from [`ConsumerConfig`][consumer-config]. Available properties for the Java Kafka consumer are described in the [documentation](http://kafka.apache.org/documentation/#consumerconfigs).
+
+### Default Settings
+
+The following Java Kafka consumer properties are overridden by default.
+
+- `auto.offset.reset` is set to `none`, to avoid the surprising `latest` default.
+
+- `enable.auto.commit` is set to `false`, since offset commits are managed manually.
+
+Use `withAutoOffsetReset` and `withEnableAutoCommit` to change these properties.
+
+In addition, there are several settings specific to the library.
+
+- `withCloseTimeout` controls the timeout when waiting for consumer shutdown. Default is 20 seconds.
+
+- `withCommitRecovery` defines how offset commit exceptions are recovered. See [`CommitRecovery.Default`][commitrecovery-default].
+
+- `withCommitTimeout` sets the timeout for offset commits. Default is 15 seconds.
+
+- `withCreateConsumer` changes how the underlying Java Kafka consumer is created. The default merely creates a Java `KafkaConsumer` instance using set properties, but this function allows overriding the behaviour for e.g. testing purposes.
+
+- `withExecutionContext` sets the `ExecutionContext` on which blocking Java Kafka consumer functions are executed. Unless specified, a default fixed single-thread pool is created as part of consumer initialization, with the thread name using the `fs2-kafka-consumer` prefix.
```

Comment:
for my curiosity, why does it use a fixed single-thread for blocking IO? I though the general rules was to use a cached thread pool for blocking IO? Is it because of order guarantees?

Reply:
It's because we know we need exactly one thread for the blocking IO. If the user already has a cached thread pool for blocking IO, they can set to use it with `withExecutionContext`.